applications/GeoMechanicsApplication/custom_strategies/strategies/README.md

```diff
@@ -0,0 +1,93 @@
+# GeoMechanicsNewtonRaphsonErosionProcessStrategy class.
+
+This class implements a Newton-Raphson process for the backward erosion piping. The physics of the piping process is
+described in
+
+1. van Beek M. Backward erosion piping,initiation and progression. TUDelft PhD. Thesis, 2015.
+2. Technisch rapport Zandmeevoerende wellen. Technische Adviescommissie
+   voor de Waterkeringen, Delft, The Netherlands, 1999.
+
+## Algorithm
+
+The algorithm is implemented in FinalizeSolutionStep function that loops over all piping elements until the erosion
+growth is finished or all piping elements are checked.
+
+### FinalizeSolutionStep function
+
+![FinalizeSolutionStep.svg](FinalizeSolutionStep.svg)
+
+This function calls other functions that are described below.
+
+### GetPipingElements function
+
+This function returns a list of all piping elements. To create the list, all elements are checked for  
+PIPE_START_ELEMENT property that holds Id of the piping starting element.
+Then elements on the list are sorted based on X-coordinate of the element begin.
+The current limitation is the following. The starting piping element has to have minimum or maximum value of the X
+coordinate. This is needed to find a direction for the piping growth.
+
+### InitialiseNumActivePipeElements function
+
+It checks the list of the elements and returns a number of elements with PIPE_ACTIVE property At the beginning the
+number is zero.
+
+### CalculateMaxPipeHeight function
+
+It calculates a maximum diameter using CalculateParticleDiameter function then multiplies it by a magic factor of 100 to
+get the height value.
+
+### CalculateParticleDiameter
+
+In case of PIPE_MODIFIED_D, the function returns
+
+$$ diameter = 2.08\times 10^{-4} * \Bigg(\frac{D_{70}}{2.08\times 10^{-4}}\Bigg)^{0.4}$$
+
+otherwise, $$diameter = D_{70}$$
+
+### check_pipe_equilibrium function
+
+This function performs iterative process with a prescribed maximum number of iterations. For each iteration the
+modelling is
+performed. Then a pipe height is calculated for each open piping element.
+
+![check_pipe_equilibrium.svg](check_pipe_equilibrium.svg)
+
+### Recalculate function
+
+This function performs the modeling using prescribed GeoMechanicsNewtonRaphsonStrategy class.
+
+![Recalculate.svg](Recalculate.svg)
+
+### CalculateEquilibriumPipeHeight function
+
+This is a function from SteadyStatePwPipingElement. It calculates the equilibrium pipe height as
+
+$$F \frac{\pi }{3} D_p \Big(\frac{\rho_s}{\rho_w} - 1\Big) \eta
+\frac{\sin(\theta + \alpha)}{\cos(\theta)  \frac{dh}{dx}}$$
+
+where $F$ is the model factor,\
+$D_p$ is the particle diameter calculated with CalculateParticleDiameter function,\
+$\rho_s$ is the solid density,\
+$\rho_w$ is the water density,\
+$\eta=\frac{n d^2}{A}$ is the packing coefficient?\
```

Comment:
What are the $n$, $d$ and $A$ here? I can speculate some 'logical' definitions, but not too sure.

Reply:
The formula is taken from Ref. 1 and I put ? for Wjitze Pieter. The coefficient is the user input, so I removed the formula. n is the number of particles, A is the unit area, and I guess d is the particle diameter.